### PR TITLE
[amazonechocontrol] Add features to control smart devices connected to the Amazon Echo

### PR DESCRIPTION
**Pull request by lkn94**
Add smart home functions to the amazonechocontrol binding. Users are able to control smart devices connected to Amazon Alexa:

- turn on/off your lights
- change the color
- control groups of lights or just single bulbs
- receive the current state of the lights
- turn on/off smart plugs

A first beta version of these functions can be found in the [openhab community](https://community.openhab.org/t/preview-and-beta-amazon-echo-control/81721).

### CODE REVIEW

**Comment by mgeramb** [comment]
There are two merge conflicts. If you need help for the merge, please let it know.

**Comment by mgeramb** [comment]
@kaikreuzer, @davidgraeff, @wborn this is a nice and big new feature for the binding, it would be nice if we can get a first review of one of the openhab experts because it is a lot of new code.

**Comment by mgeramb** [comment]
Looks now good to me, thank you!

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/smarthome/SmartHomeDeviceStateGroupUpdateCalculator.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+/**
+ * Handles the update interval calculation
+ *
+ * @author Michael Geramb
+ */
+@NonNullByDefault
+public class SmartHomeDeviceStateGroupUpdateCalculator {
+    static final boolean isDebug = java.lang.management.ManagementFactory.getRuntimeMXBean().getInputArguments()
+            .toString().indexOf("-agentlib:jdwp") > 0;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/smarthome/SmartHomeDeviceStateGroupUpdateCalculator.java]: You shouldn't be accessing jvm params in the bindings, if you want to do this do it through a config instead.

File: bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/smarthome/SmartHomeDevicesDiscovery.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Lukas Knoeller
+ */
+@NonNullByDefault
+public class SmartHomeDevicesDiscovery extends AbstractDiscoveryService implements ExtendedDiscoveryService {
+
+    AccountHandler accountHandler;
```
Comment (cpmeister) [inline bundles/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/smarthome/SmartHomeDevicesDiscovery.java]: Please make sure to make fields private if possible.

**Comment by J-N-K** [comment]
@lkn94, are willing to continue here? I could provide assistance, if needed.

**Comment by lkn94** [comment]
> @lkn94, are willing to continue here? I could provide assistance, if needed.

I wanna continue but I don't have enough time at the moment. So the binding must wait ;)

**Pull request closed without merging**